### PR TITLE
Replace Versioned interface with ERC165 support

### LINKED ISSUE #240: Replace ITempusPool.version with ERC-165 support

### PR DESCRIPTION
**Pull request by pashov**
Fixes #240

### CODE REVIEW

File: contracts/TempusPool.sol [hunk 1]
```diff
@@ -618,4 +617,8 @@ abstract contract TempusPool is ITempusPool, ReentrancyGuard, Ownable, Versioned
 
     /// @return Converts an interest rate decimal into a Principal/Yield Share decimal
     function interestRateToSharePrice(uint256 interestRate) internal view virtual returns (uint256);
+
+    function supportsInterface(bytes4 interfaceId) external view override returns (bool) {
+        return interfaceId == type(IERC165).interfaceId || interfaceId == type(ITempusPool).interfaceId;
```
Comment (mijovic) [inline contracts/TempusPool.sol:622]: Isn't this enough? `ITempusPool` inherits `IERC165` already
```suggestion
        return interfaceId == type(ITempusPool).interfaceId;
```

File: test/TempusPool.Deploy.test.ts [hunk 2]
```diff
@@ -93,4 +93,16 @@ describeForEachPool("TempusPool Deploy", (testPool:PoolTestFixture) =>
     (await expectRevert(testPool.create({ initialRate:1.0, poolDuration:60, yieldEst:0 })))
       .to.equal(":ZeroEstimatedFinalYield");
   });
+
+  it.only("Should support ERC165 standard", async() =>
```
Comment (mijovic) [inline test/TempusPool.Deploy.test.ts:97]:
```suggestion
  it("Should support ERC165 standard", async() => 
```

**Comment by mijovic** [comment]
You will just need to rebase to get the latest changes from master

**Review by mijovic — APPROVED** [review]

File: contracts/TempusController.sol [hunk 3]
```diff
@@ -578,4 +577,8 @@ contract TempusController is ITempusController, ReentrancyGuard, Ownable, Versio
             return _redeemToYieldBearing(tempusPool, address(this), principals, yields, msg.sender);
         }
     }
+
+    function supportsInterface(bytes4 interfaceId) external view override returns (bool) {
+        return interfaceId == type(ITempusController).interfaceId;
```
Comment (axic) [inline contracts/TempusController.sol:582]: This has to extend `super` imho, just to be compatible with ERC-165.

From OZ:
```
  * Contracts that want to implement ERC165 should inherit from this contract and override {supportsInterface} to check
 * for the additional interface id that will be supported. For example:
 *
 * ```solidity
 * function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
 *     return interfaceId == type(MyInterface).interfaceId || super.supportsInterface(interfaceId);
 * }
```

Comment (pashov) [reply contracts/TempusController.sol:582]: Addressed here: https://github.com/tempus-finance/tempus-protocol/pull/630/files

**Pull request merged**